### PR TITLE
transfered appContentSwitcher to ts

### PR DESCRIPTION
**Pull request by greentoss**
transfered appContentSwitcher to ts

### CODE REVIEW

File: src/components/app-content-switcher/AppContentSwitcher.tsx [hunk 1]
```diff
@@ -1,12 +1,36 @@
...
+    styles?: object;
+}
+
+type SwitchOption = {
+    text: string,
+    tooltip?: string
+}
+
+const AppContentSwitcher = ({ active, handleChange, switchOptions, typographyVariant, styles }: SwitcherProps) => {
+
```
Comment (Yurenko) [inline src/components/app-content-switcher/AppContentSwitcher.tsx:32]: If you want to describe type for props, you need to use another syntax.
Like this: const AppContentSwitcher: FC<SwitcherProps> = ({ active, handleChange, switchOptions, typographyVariant, styles }) =>

But first of all import {FC} from 'react'

File: src/components/app-content-switcher/AppContentSwitcher.tsx [hunk 2]
```diff
@@ -1,12 +1,36 @@
...
-  const renderBlock = (options, active) => (
+interface SwitcherProps {
+    active: boolean;
+    handleChange: () => void;
+    switchOptions: {
+        left?: {
+            text: string;
+            tooltip?: string;
+        };
+        right?: {
```
Comment (Tolik170) [inline src/components/app-content-switcher/AppContentSwitcher.tsx:17]: type declaration for left and right is the same, create a type and use it, also can create type for switchOptions:

type switchOption = {
test: string
tooltip?: string
}
type switcOptions = {
left?:  someType
right: someType
}

But, think about better naming

File: src/components/app-content-switcher/AppContentSwitcher.tsx [hunk 3]
```diff
@@ -1,12 +1,36 @@
...
+            text: string;
+            tooltip?: string;
+        };
+    };
+    typographyVariant: TypographyProps['variant'];
+    styles?: object;
+}
+
+type SwitchOption = {
```
Comment (Tolik170) [inline src/components/app-content-switcher/AppContentSwitcher.tsx:26]: you allready have this type, so you can use it in the interface above

File: src/components/app-content-switcher/AppContentSwitcher.tsx (hunk 1 above)
Comment (greentoss) [reply src/components/app-content-switcher/AppContentSwitcher.tsx]: refactored

File: src/components/app-content-switcher/AppContentSwitcher.tsx (hunk 2 above)
Comment (greentoss) [reply src/components/app-content-switcher/AppContentSwitcher.tsx]: refactored

File: src/components/app-content-switcher/AppContentSwitcher.tsx (hunk 3 above)
Comment (greentoss) [reply src/components/app-content-switcher/AppContentSwitcher.tsx]: refactored

File: src/components/app-content-switcher/AppContentSwitcher.tsx [hunk 4]
```diff
@@ -1,12 +1,31 @@
+import { TypographyProps } from '@mui/material/Typography'
 import { Typography } from '@mui/material'
 import Tooltip from '@mui/material/Tooltip'
 import Switch from '@mui/material/Switch'
 import Stack from '@mui/material/Stack'
 import { defaultStyles } from '~/components/app-content-switcher/AppContentSwitcher.styles'
+import { FC } from 'react'
```
Comment (Yurenko) [inline src/components/app-content-switcher/AppContentSwitcher.tsx:7]: Put this import first in this list

Comment (greentoss) [reply src/components/app-content-switcher/AppContentSwitcher.tsx]: fixed

**Review by maarekua — APPROVED** [review]

File: src/components/app-content-switcher/AppContentSwitcher.tsx [hunk 5]
```diff
@@ -1,12 +1,23 @@
...
 import { defaultStyles } from '~/components/app-content-switcher/AppContentSwitcher.styles'
+import { SwitchContent, SwitchOptions } from '~/types'
 
-const AppContentSwitcher = ({ active, handleChange, switchOptions, typographyVariant, styles = {} }) => {
+interface SwitcherProps {
+    active: boolean;
+    onChange: () => void;
+    switchOptions: SwitchOptions
+    typographyVariant: TypographyProps['variant'];
+    styles?: object;
```
Comment (Tolik170) [inline src/components/app-content-switcher/AppContentSwitcher.tsx:15]: use SxProps for styles

Comment (greentoss) [reply src/components/app-content-switcher/AppContentSwitcher.tsx]: fixed

**Review by Tolik170 — APPROVED** [review]

**Pull request merged**